integration/spark-common/src/main/scala/org/apache/spark/util/PartitionUtils.scala

```diff
@@ -113,7 +113,8 @@ object PartitionUtils {
     }
 
     if (partitionId == 0) {
-      partitionInfo.addPartition(splitInfo.size)
+      val addedPartitionList = PartitionUtils.getListInfo(splitInfo.mkString(","))
```

Comment:
Looks like `add partition` of list partition table can only support string, but not string group. 
If need to support string group, splitInfo should be List[List[String]] instead of List[String].
Could you also change other places to support this feature and add test case?

Reply:
Add partition does support String group. The problem in this bug is, when you are trying to add these two partitions "'SAUDI ARABIA,(VIETNAM,RUSSIA,UNITED KINGDOM,UNITED STATES)'" then splitInfo variable takes it as one partition and increment partition count by 1. To get the correct partitions out of it, we need to extract the list out of the splitList.

As this is a sequence based test case, where the problem is encountered when we  "show partition" after "alter table to add partition", So is it alright to add its test case to TestAlterPartitionTable.scala?